src/lib/network/network_autoconfiguration.rb

```diff
@@ -91,6 +91,10 @@ def configure_dhcp
       end
 
       activate_changes(dhcp_cards.map(&:name))
+
+      # Force a read of the configuration just for reading the transient
+      # hostname as it could be modified through dhcp since previous read.
+      Lan.read_config
```

Comment:
NP or kind of ~~brain~~storming: in theory it makes more sense to re-read dhcp hostname before displaying it to the user then after activatiing changes. Bcs user could be impressed by YaST UI and kept it open for several hours and the transient host name might change in the meantime ....

Reply:
Yep, in fact that is the problem about storing the transient **hostname** when it is set by **dhcp**, it can be modified dynamically. But as currently it is the static one which is shown then it is not a problem.

The only problem could be the proposal of the hostname, but, as we are now not proposing a random hostname at all, now should be ok.

Reply:
yast should mostly deal with static hostname mostly. Only case when we need transient hostname is when we use it as a proposal for static hostname. So, we mostly don't need to bother with transient hostname.

Reply:
Exactly, and if we still want to keep that proposal, then I would do it separately. The transient hostname could be added to the dialog, in order to show the current one, but would do it dynamically instead of storing it as part of the config.

Reply:
yes, I agree. So the only point is whether we need to enforce re-reading transient hostname after write ... however, I assume it is cheap so keep it as it is.